### DISCUSSION TITLE
Search function collides Japanese IME in Enter/Esc keys

### DISCUSSION THREAD

**Discussion by amane-katagiri** [requarks/wiki#3762] [Error / Bug Report]
**Describe the bug**

I confirm / cancel the conversion with Japanese IME, the search bar will also confirm / cancel at the same time, so I cannot enter Japanese correctly.

`keyUp` event will return keyCode=13 (Enter) always. But `keyDown` event during Japanese IME conversion will not return keyCode=13 (Enter).

**To Reproduce**

1. Input "にほんご" (nihongo) with Japanse IME to the search bar.
1. Convert "にほんご" to "日本語" with pressing space.
1. Press Enter.
1. If some articles found, the first article shows immediately.
1. If no article found, the search bar closes immediately.

**Expected behavior**

1. Input "にほんご" (nihongo) with Japanse IME to the search bar.
1. Convert "にほんご" to "日本語" with pressing space.
1. Press Enter.
1. The search bar remains in the input acceptance state (you can input any other words).
1. Confirm all input, and press enter.
1. If some articles found, the first article shows immediately.
1. If no article found, the search bar closes immediately.

**Screenshots**

To Reproduce 1:
![1](https://user-images.githubusercontent.com/8361232/112375828-1629a100-8d27-11eb-9b85-08eefe557cd6.png)

To Reproduce 2:
![2](https://user-images.githubusercontent.com/8361232/112375831-16c23780-8d27-11eb-8737-4e8d3e189a28.png)

To Reproduce 3 and 5:
![3](https://user-images.githubusercontent.com/8361232/112375832-175ace00-8d27-11eb-8422-636abbedf662.png)

Host Info (please complete the following information):
OS: Alpine and Docker (requarks/wiki:2)
Wiki.js version: 2.5.191
Database engine: PostgreSQL 11.11

**Comment by kurank** [comment]
I had the same problem on v2.5.307: pressing Enter to confirm IME composition (likely not only for Japanese, but also for users of Chinese, Korean, and other East Asian input methods) always triggers a searchEnter event and immediately navigates to the first result.


**Fix**
In `wiki/client/components/common/nav-header.vue` → `methods: { searchEnter(event) { … } }` add:
```js
if (event.isComposing || event.keyCode === 229) {
  return;
}
```

before the rest of the handler, so that IME composition-confirm Enter is ignored.

**Tested**
- Confirmed Japanese input (“にほんご” → “日本語”) with IME: no premature navigation
- Normal Enter (no IME): still triggers search as expected

Would you consider including this check upstream? Thanks!
